### DISCUSSION TITLE
classes/class_editorproperty

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#224] [User-contributed notes]
# classes/class_editorproperty

Inherits: Container< Control< CanvasItem< Node< Object Custom control for editing properties that can be added to the EditorInspector. Description: A custom control for editing properties that can ...

https://docs.godotengine.org/en/stable/classes/class_editorproperty.html

**Comment by Bogost** [comment]
get_edited_object() and get_edited_property() will return null if used before _ready() phase of EditorProperty.

**Reply by guicattani** [reply]
Thank you for taking the time to write this <3 this helped me a ton

**Reply by Calinou** [reply]
Pull request opened:

- https://github.com/godotengine/godot/pull/106807

**Comment by elliotfontaine** [comment]
What's the point of `use_folding`? Is it to keep the folding across editor sessions / inspected object switch?

**Reply by ajreckof** [reply]
use folding is a bit of a strange case as it is not really used by editor properties per say but more to be passed on to inner elements that will use them. By default `use_folding` is set to true and when not inspector group are always be open and cannot be folded. Typically this is usefull for remote where tfolding is always disabled but also some people prefer there inspector that way so there is an editor settings to disable folding everywhere. 
Now on why all of this means that EditorProperty needs this information. When an inspector is created they cannot by themself know whethere they are editing or not a remote object. Therefore this information has to be passed down from the main inspector to all sub-inspector and this goes through `EditorProperty`. This is mostly for `EditorPropertyResource` but is usefull for any other `EditorProperty` that can include an `EditorInspector`, including all those that can include `EditorPropertyResource` such as arrays and dictionaries.

What is on the other quite strange is that use_folding was exposed here but not on `EditorInspector`.
